### PR TITLE
Add MapDatasetEventSampler sample_edisp method

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR aims at adding a `sample_edisp` method in the `MapDatasetEventSampler` class. The method applies energy dispersion correction to the energies of the simulated events.

### CODE REVIEW

File: gammapy/cube/simulate.py [hunk 1]
```diff
@@ -176,3 +176,33 @@ def sample_background(self, dataset):
...
+            {
+                "lon": events.table["RA"].quantity,
+                "lat": events.table["DEC"].quantity,
+                "energy": events.table["ENERGY_TRUE"].quantity,
+            },
+            coordsys="icrs",
+        )
+
+        coords_reco = edisp_map.sample_coord(coord, self.random_state)
+        events.table["ENERGY_TRUE"] = coords_reco["energy"] * u.TeV
```
Comment (adonath) [inline gammapy/cube/simulate.py]: I think you can just use `table["ENERGY"] = coords_reco["energy"] * u.TeV`

File: gammapy/cube/tests/test_simulate.py [hunk 2]
```diff
@@ -116,3 +117,19 @@ def test_MDE_sample_background():
...
+def test_MDE_sample_edisp():
+    edisp_map = make_edisp_map_test()
+
+    dataset = dataset_maker()
+    sampler = MapDatasetEventSampler(random_state=0)
+    events = sampler.sample_sources(dataset=dataset)
+    events = sampler.sample_edisp(edisp_map, events)
+
+    assert len(events.table) == 726
+    assert_allclose(events.table["ENERGY_TRUE"][0], 1.0600765557667795, rtol=1e-5)
```
Comment (adonath) [inline gammapy/cube/tests/test_simulate.py]: Can you add an additional assert on the unit of the columns, like I did here: https://github.com/gammapy/gammapy/pull/2659/commits/7c94768ac05b548c818ed4af17cfe46f72a1ed75

File: gammapy/cube/simulate.py (hunk 1 above)
Comment (fabiopintore) [reply gammapy/cube/simulate.py]: Done

File: gammapy/cube/tests/test_simulate.py (hunk 2 above)
Comment (fabiopintore) [reply gammapy/cube/tests/test_simulate.py]: Done.

**Comment by adonath** [comment]
Thanks @fabiopintore!

**Pull request merged**